modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgnitePdsRecoveryAfterFileCorruptionTest.java

```diff
@@ -175,8 +190,42 @@ public void testPageRecoveryAfterFileCorruption() throws Exception {
         stopAllGrids();
 
         ig = startGrid(0);
+        ig.cluster().state(ClusterState.ACTIVE);
+
+        checkRestore(ig, pages);
+
+        // It is necessary to clear the current WAL history to make sure that the restored pages have been saved.
+        GridCacheSharedContext<Object, Object> cctx = ig.context().cache().context();
+        GridCacheDatabaseSharedManager dbMgr = (GridCacheDatabaseSharedManager)cctx.database();
+        FileWriteAheadLogManager wal = (FileWriteAheadLogManager)cctx.wal();
+
+        dbMgr.enableCheckpoints(true).get(getTestTimeout());
+
+        WALPointer trucateWalPtr = dbMgr.checkpointHistory().lastCheckpoint().checkpointMark();
+
+        IgniteCache<Object, Object> dummyCache = ig.cache(dummyCacheName);
 
-        ig.cluster().active(true);
+        byte[] dummyData = new byte[WAL_SEGMENT_SIZE / 2];
+
+        // Try to move current WAL segment into archive.
+        for (int i = 0; i < WAL_SEGMENTS_CNT * 2; i++) {
+            dummyCache.put(i, dummyData);
```

Comment:
`wal.log(new CheckpointRecord(null), RolloverType.NEXT_SEGMENT);` instead of cache puts?

Reply:
Done, thank you!